llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -863,6 +863,25 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     }
     break;
   }
+  case Intrinsic::get_active_lane_mask: {
+    if (ST->hasVInstructions()) {
+      Type *ExpRetTy = VectorType::get(
+          ICA.getArgTypes()[0], cast<VectorType>(RetTy)->getElementCount());
+      auto LT = getTypeLegalizationCost(ExpRetTy);
+
+      // vid.v   v8
```

Comment:
`vid.v v8` will always be hoisted I think, should we take this into consideration?

Reply:
I thought for that case we may need an argument to know if the user want to include `vid.v` in the cost.
However upon reviewing BasicTTIImpl.h, I notice that the cost is calculated as a sum of  `uadd_sat` and `ICmp`.
I think I can follow that approach.